server/dialect-idms/src/main/java/org/eclipse/lsp/cobol/dialects/idms/IdmsDialect.java

```diff
@@ -206,7 +209,8 @@ private Collection<SyntaxError> getWarningForNoAdjustment(
       Pair<Range, Integer> rangeIntegerPair,
       String uri,
       Locality sourceLocality) {
-    if (shouldLevelNotBeAdjusted(copybookLevel, firstLevel, rangeIntegerPair.getRight())) {
+    if (shouldLevelNotBeAdjusted(copybookLevel, firstLevel, rangeIntegerPair.getRight())
+        && !SPECIAL_LEVELS.contains(rangeIntegerPair.getRight())) {
```

Comment:
It's better to move it into the shouldLevelNotBeAdjusted method.

Reply:
Updated